src/main/java/org/apache/commons/text/similarity/SorensenDiceSimilarity.java

```diff
@@ -0,0 +1,146 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.text.similarity;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * A similarity algorithm indicating the percentage of matched characters
+ * between two character sequences.
+ *
+ * <p>
+ * The S&#248;rensen-Dice coefficient is a statistic used for comparing the
+ * similarity of two samples. It was independently developed by the botanists
+ * Thorvald S&#248;rensen and Lee Raymond Dice, who published in 1948 and 1945
+ * respectively. The index is known by several other names, especially
+ * S&#248;rensen-Dice index, S&#248;rensen index and Dice's coefficient. Other
+ * variations include the "similarity coefficient" or "index", such as Dice
+ * similarity coefficient (DSC).
+ * </p>
+ *
+ * <p>
+ * This implementation is based on the S&#248;rensen-Dice similarity algorithm
+ * from <a href=
+ * "http://en.wikipedia.org/wiki/S%C3%B8rensen%E2%80%93Dice_coefficient">
+ * http://en.wikipedia.org/wiki/S%C3%B8rensen%E2%80%93Dice_coefficient</a>.
+ *
+ *
+ * </p>
+ *
+ * @since 1.7
+ */
+public class SorensenDiceSimilarity implements SimilarityScore<Double> {
+
+    /**
+     * For shifting bigrams to fit in single integer.
+     */
+    private static final int SHIFT_NUMBER = 16;
+
+    /**
+     * Converter function for conversion of string to bigrams.
+     */
+    private final Function<CharSequence, Collection<Integer>> converter = new SorensenDiceConverter();
+
+    /**
+     * Measures the overlap of two sets created from a pair of character sequences.
+     * {@link OverlapSimilarity}}
+     */
+    private final IntersectionSimilarity<Integer> similarity = new IntersectionSimilarity<>(this.converter);
+
+    /**
+     * Calculates Sorensen-Dice Similarity of two character sequences passed as
+     * input.
+     *
+     * <pre>
+     * similarity.apply(null, null)                 = IllegalArgumentException
+     * similarity.apply("foo", null)                = IllegalArgumentException
+     * similarity.apply(null, "foo")                = IllegalArgumentException
+     * similarity.apply("night", "nacht")           = 0.25
+     * similarity.apply("", "")                     = 1.0
+     * similarity.apply("foo", "foo")               = 1.0
+     * similarity.apply("foo", "foo ")              = 0.8
+     * similarity.apply("foo", "foo ")              = 0.66
```

Comment:
There are some erroneous examples in the docstring. `similarity.apply("foo", "foo ")` is repeated twice, and the second result (0.66) is incorrect. 
Likewise, `similarity.apply("foo", " foo ")` should be 0.66666...

Reply:
ping @ameyjadiye 

Good spotting @mvarela